components/blitz/src/omero/gateway/facility/ROIFacility.java

```diff
@@ -540,6 +536,23 @@ public ROIResult loadROI(SecurityContext ctx, long roiId)
             IUpdatePrx updateService = gateway.getUpdateService(ctx);
             IRoiPrx svc = gateway.getROIService(ctx);
             
+            Collection<ROIData> updated = new ArrayList<ROIData>();
+
+            if (imageID < 0) {
+                for (ROIData r : roiList) {
+                    if (r.getId() > -1)
+                        throw new OperationNotSupportedException(
```

Comment:
odd choice -- this exception is related to `javax.naming.Context` -- maybe you meant `UnsupportedOperationException`? Also Apache commons has a `NotImplementedException`.